### PR TITLE
Fix Static compile for FIPS

### PR DESCRIPTION
**Pull request by zulfilee**
FIx the static compile part for FIPS based compilation

### CODE REVIEW

**Review by jayesh-srivastava — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api/pull/106#pullrequestreview-1499041523" title="Approved">jayesh-srivastava</a>*, *<a href="https://github.com/spectrocloud/cluster-api/pull/106#" title="Author self-approved">zulfilee</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign  after the PR has been reviewed.
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api/blob/spectro-release-4.0/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
